### PR TITLE
use jq schema for content_key in json_loader

### PR DESCRIPTION
**Pull request by kzk-maeda**
### Description
Changed the value specified for `content_key` in JSONLoader from a single key to a value based on jq schema.

Sorry, I created the [same Pull Request](https://github.com/langchain-ai/langchain/pull/8075) in the past, but some time has elapsed and it was closed, so I am recreating it.

### Why
For json data like the following, specify `.data[].attributes.message` for page_content and `.data[].attributes.id` or `.data[].attributes.attributes. tags`, etc., the `content_key` must also parse the json structure.

<details>
<summary>sample json data</summary>

```json
{
  "data": [
    {
      "attributes": {
        "attributes": {
          "dd": {
            "service": "worker",
            "env": "production",
            "version": "6b2c46a5883a9097aa2cb09907786b5c06ca3bd0"
          },
          "source": "stderr",
          "service": "worker",
          "name": "app.services.predict_services",
          "levelname": "ERROR",
          "container_id": "f9f880b243cc41f1a7d9bae5bf922d60-1262558729",
          "timestamp": 1686679407827.0
        },
        "message": "error while processing user #19084: 'numpy.dtype[bool_]' object is not callable",
        "service": "worker",
        "status": "error",
        "tags": [
          "datadog.submission_auth:private_api_key",
          "env:production"
        ],
        "timestamp": "2023-06-14T03:03:27.827000+09:00"
      },
      "id": "AgAAAYi17UzTUeIczgAAAAAAAAAYAAAAAEFZaTE3VThXQUFEOF9sS2J4Z3psRmdBRAAAACQAAAAAMDE4OGI2MzYtMmZlNC00ZDEwLThjZDMtMzhkZTI0NmUyNWMz",
      "type": "log"
    },
    {
      "attributes": {
        "attributes": {
          "dd": {
            "service": "worker",
            "env": "production",
            "version": "6b2c46a5883a9097aa2cb09907786b5c06ca3bd0"
          },
          "process": 42.0,
          "messages": "error while processing user #19084: 'numpy.dtype[bool_]' object is not callable",
          "levelname": "ERROR",
          "container_id": "f9f880b243cc41f1a7d9bae5bf922d60-1262558729",
          "timestamp": 1686679407831.0
        },
        "message": "{\"messages\": \"error while processing user #19084: 'numpy.dtype[bool_]' object is not callable\"}",
        "service": "worker",
        "status": "error",
        "tags": [
          "datadog.submission_auth:private_api_key",
          "env:production"
        ],
        "timestamp": "2023-06-14T03:03:27.831000+09:00"
      },
      "id": "AgAAAYi17UzXUeIczwAAAAAAAAAYAAAAAEFZaTE3VThXQUFEOF9sS2J4Z3psRmdBRQAAACQAAAAAMDE4OGI2MzYtMmZlNC00ZDEwLThjZDMtMzhkZTI0NmUyNWMz",
      "type": "log"
    }
  ],
  "meta": {
    "elapsed": 26,
    "request_id": "pddv1ChY0SmttaDA0a1REeXZRM01yNkFwYnd3Ii0KHWIANr8mghGpsMIX2cOarI6t4WyTVObXx3wrAuudEgzSbtmduLtPxFVkSo0",
    "status": "done"
  }
}

```

</details>

<details>
<summary>sample code</summary>

```python
def metadata_func(record: dict, metadata: dict) -> dict:
    print(record)

    metadata["id"] = record.get("id")
    metadata["tags"] = record["attributes"].get("tags")

    return metadata

sample_file = "sample.json"
loader = JSONLoader(
    file_path=sample_file,
    jq_schema=".data[]",
    content_key=".attributes.message",
    metadata_func=metadata_func
)
```

</details>

### Dependencies
none

### Tag maintainer
@rlancemartin, @eyurtsev

### Twitter handle
[kzk_maeda](https://twitter.com/kzk_maeda)

### CODE REVIEW

File: libs/langchain/langchain/document_loaders/json_loader.py [hunk 1]
```diff
@@ -49,7 +50,7 @@ def __init__(
 
         self.file_path = Path(file_path).resolve()
         self._jq_schema = jq.compile(jq_schema)
-        self._content_key = content_key
+        self._content_key = jq.compile(content_key) if content_key else None
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/json_loader.py:53]: The current implementation was added in May, so there might be existing users relying on the previous functionality.

Would you be able to make this change backwards compatible? We could introduce a new attribute that if provided will take precedence

Comment (kzk-maeda) [reply libs/langchain/langchain/document_loaders/json_loader.py:53]: Understood, I will think about it a bit.

Comment (kzk-maeda) [reply libs/langchain/langchain/document_loaders/json_loader.py]: @eyurtsev 
To ensure backward compatibility, I've added a flag named `is_content_key_jq_parsable`, which is set to False by default. 
When `is_content_key_jq_parsable=False`, you can specify `content_key` as a string, as before. 
When `is_content_key_jq_parsable=True`, you can specify a jq schema key for `content_key`.

**Comment by baskaryan** [comment]
Apologies for the slow review! Pr has some merge conflicts, happy to re-review if you'd like to resolve

**Comment by kzk-maeda** [comment]
I re-created new PR: https://github.com/langchain-ai/langchain/pull/18003
and I'm going to close this PR.

**Pull request closed without merging**